packages/discord.js/src/structures/interfaces/TextBasedChannel.js

```diff
@@ -165,6 +165,11 @@ class TextBasedChannel {
       return dm.send(options);
     }
 
+    if (!options.nonce && this.client.options.automatedMessageNonce) {
```

Comment:
A nonce can be `0`, so this may incorrectly pass the check.

Reply:
Updated